src/main/java/org/fundacionjala/coding/melvi/FizzBuzz.java

```diff
@@ -0,0 +1,79 @@
+package org.fundacionjala.coding.melvi;
+
+
+/**
+ * Imagine the scene. You are eleven years old, and in the five minutes
+ * before the end of the lesson, your Maths teacher
+ * decides he should make his class more “fun” by introducing a “game”.
+ * He explains that he is going to point at each
+ * pupil in turn and ask them to say the next number in sequence,
+ * starting from one. The “fun” part is that if the
+ * number is divisible by three, you instead say “Fizz” and if it is
+ * divisible by five you say “Buzz”. So now your
+ * maths teacher is pointing at all of your classmates in turn, and
+ * they happily shout “one!”, “two!”, “Fizz!”, “four!”,
+ * “Buzz!”… until he very deliberately points at you, fixing you with
+ * a steely gaze… time stands still, your mouth dries
+ * up, your palms become sweatier and sweatier until you finally manage to
+ * croak “Fizz!”. Doom is avoided, and the pointing
+ * finger moves on.
+ * <p>
+ * So of course in order to avoid embarassment infront of your whole class,
+ * you have to get the full list printed out so
+ * you know what to say. Your class has about 33 pupils and he might go
+ * round three times before the bell rings for breaktime.
+ * Next maths lesson is on Thursday. Get coding!
+ * <p>
+ * Write a program that prints the numbers from 1 to 100. But for multiples
+ * of three print “Fizz” instead of the number
+ * and for the multiples of five print “Buzz”. For numbers which are
+ * multiples of both three and five print “FizzBuzz “.
+ */
+public final class FizzBuzz {
+    public static final int LIMIT = 101;
+    public static final int THREE = 3;
+    public static final int FIVE = 5;
+    private static final String FIZZ = "Fizz";
+    private static final String BUZZ = "Buzz";
+
+    /**
+     * constructor.
+     */
+    protected FizzBuzz() {
+
+    }
+
+    /**
+     * Print.
+     * @return string
+     */
+    public String printFizzBuzz() {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 1; i < LIMIT; i++) {
+            sb.append(processFizzBuzz(i));
+            sb.append(" ");
+        }
+        return sb.toString();
+    }
+
+    /**
+     * @param number int number
+     * @return String Fizz, Buzz or FizzBuzz
+     */
+    public String processFizzBuzz(int number) {
+        String result = "";
+        if (number % THREE == 0 || String.valueOf(number).contains("3")) {
+            result = FIZZ;
+
+        }
+        if (number % FIVE == 0 || String.valueOf(number).contains("5")) {
+            result = result.concat(BUZZ);
+        }
+        if (result.equals("")) {
+            result = String.valueOf(number);
+        }
+        return result;
```

Comment:
This may be refactored like this.
```java
        if (number % (FIVE * THREE) == 0) {
            return FIZZ.concat(BUZZ);
        }
        String numberAsString = String.valueOf(number);
        if (number % FIVE == 0 || numberAsString.contains("5")) {
            return BUZZ;
        }
        if (number % THREE == 0 || numberAsString.contains("3")) {
            return FIZZ;
        }
        return numberAsString;
```